app/assets/javascripts/Api.elm

```diff
@@ -60,9 +60,22 @@ searchMore wmsg result =
             Task.perform (wmsg << SearchFailed) (wmsg << SearchSuccess) (Http.get Decoders.search nextUrl)
 
 
+type FacilityTypesMsg
+    = FacilityTypesSuccess (List FacilityType)
+    | FacilityTypesFailed Http.Error
+
+
+getFacilityTypes : (FacilityTypesMsg -> msg) -> Cmd msg
+getFacilityTypes wmsg =
+    let
+        url = "/api/facility_types"
+    in
+        Task.perform (wmsg << FacilityTypesFailed) (wmsg << FacilityTypesSuccess) (Http.get Decoders.facilityTypes url)
+
+
 emptySearch : SearchSpec
 emptySearch =
-    { q = Nothing, s = Nothing, l = Nothing, latLng = Nothing }
+    { q = Nothing, s = Nothing, l = Nothing, latLng = Nothing, fType = Nothing, fName = Nothing }
```

Comment:
What is the difference between `q` and `fName`? I know the API will eventually include the option to perform a text over the whole body of a facility, but I don't see a use case for that in out app.